python/packages/autogen-ext/src/autogen_ext/models/cache.py

```diff
@@ -0,0 +1,186 @@
+import hashlib
+import json
+import warnings
+from typing import Any, AsyncGenerator, List, Mapping, Optional, Sequence, Union, cast
+
+from autogen_core import CancellationToken
+from autogen_core.models import (
+    ChatCompletionClient,
+    CreateResult,
+    LLMMessage,
+    ModelCapabilities,  # type: ignore
+    ModelInfo,
+    RequestUsage,
+)
+from autogen_core.store import AbstractStore
+from autogen_core.tools import Tool, ToolSchema
+
+
+class ChatCompletionCache(ChatCompletionClient):
+    """
+    A wrapper around a ChatCompletionClient that caches creation results from an underlying client.
+    Cache hits do not contribute to token usage of the original client.
+    """
+
+    def __init__(self, client: ChatCompletionClient, store: AbstractStore):
+        """
+        Initialize a new ChatCompletionCache.
+
```

Comment:
Here we can show an example of using diskcache and perhaps another one using redis.